pkg/tracing/tracing.go

```diff
@@ -0,0 +1,57 @@
+// Package tracing implements a common tracing initialization for GD2 and CLI
+package tracing
+
+import (
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+
+	log "github.com/sirupsen/logrus"
+	flag "github.com/spf13/pflag"
+	"go.opencensus.io/exporter/jaeger"
+	"go.opencensus.io/trace"
+)
+
+// Commandline options for exporter endpoints
+const (
+	JaegerEndpointOpt      = "jaegerendpoint"
+	JaegerAgentEndpointOpt = "jaegeragentendpoint"
+)
+
+// InitFlags initializes the command line options for GD2 tracing endpoints
+func InitFlags() {
+	flag.String(JaegerEndpointOpt, "", "Jaeger collector endpoint that accepts spans from Jaeger agent.")
+	flag.String(JaegerAgentEndpointOpt, "", "Jaeger agent endpoint that the Jaeger client sends spans to.")
+}
+
+// InitJaegerExporter initializes the jaeger exporter as the tracing endpoint
+// This should be called early when a process starts.
+// This creates and returns an exporter if successful.
+// Otherwise, a warning is logged and a 'nil' exporter is returned.
+func InitJaegerExporter(jaegerEndpoint string, jaegerAgentEndpoint string) *jaeger.Exporter {
+	// Create the Opencensus Jaeger exporter
```

Comment:
As mentioned in my earlier comment, you can fetch the configured options here directly, and you need to do it once. Just  `import config "github.com/spf13/viper` and use the config package.

Create and register the exporter only if both options have been set (`!= ""). Return a nil otherwise.